decode.go

```diff
@@ -134,7 +154,10 @@ func UnmarshalTable(t *ast.Table, v interface{}) (err error) {
 		case *ast.Table:
 			fv, fieldName, found := findField(rv, key)
 			if !found {
-				return fmt.Errorf("line %d: field corresponding to `%s' is not defined in `%T'", av.Line, key, v)
+				if l := getLogger(); l != nil {
```

Comment:
What I don't want is setting a logger to side effect the parsing behavior as that is a confusing API. If we want to preserve original behavior I say we add an explicit flag for it. For simplicity I broke existing behavior since I assumed no one relied on it.